### PR TITLE
[RFC] Using "secure-" property prefix in DTS for clocks/resets/...

### PR DESCRIPTION
**Pull request by etienne-lms**
**(edited: text added, P-R moved to [RFC])**
Proposal to use precedence of `secure-` prefix in FDT property name for various resources.

As in this first P-R version: update helper functions to apply on:
`clocks`, `interrupt`, `reg`, `resets`.

### CODE REVIEW

File: core/kernel/dt.c [hunk 1]
```diff
@@ -309,3 +310,23 @@ void _fdt_fill_device_info(void *fdt, struct dt_node_info *info, int offs)
 
 	*info = dinfo;
 }
+
+const void *_fdt_get_secure_prop(const void *fdt, int nodeoffset,
+				 const char *name, int *lenp)
+{
+	const fdt32_t *cuint;
+	char sec_prop[32] = { 0 };
```
Comment (jforissier) [inline core/kernel/dt.c:318]: Minor: `sec_name` for consistency with `name` perhaps

Comment (etienne-lms) [reply core/kernel/dt.c:318]: ok

**Comment by etienne-lms** [comment]
> For commit "core: dt: helper function for secure properties"
It would be nice to link to the kernel documentation: 

You're right. I'll add it.

> For commit "core: dt: use _fdt_get_secure_prop() in DT helper functions"
Only secure-status is currently mentioned in Linux kernel bindings. Should we propose a patch to document the new ones?

The doc mentions _"any property that needs a different value in the Secure world can be supported by prefixing the property name with "secure-"_. So really need a list of supported properties? I'll post a patch in the LKML and get their feedback but I fear to be exhaustive. Maybe start from "reg", "clocks", "resets" and  "interrupts" only.

> For commit "stm32_gpio: secure-clocks property as precedence on clocks in FDT":
s/as/has/ in the subject

Ack. Thanks.

**Comment by etienne-lms** [comment]
> Only secure-status is currently mentioned in Linux kernel bindings. Should we propose a patch to document the new ones?

I posted https://lkml.org/lkml/2020/4/23/82. Presents clocks and resets.
I initially created this P-R to address resources under hardware access in secure world and scmi access in non-secure world. But now I think this should not be addressed in the DT. 

That said, this P-R maybe still be of interest but maybe useless in practice. I'll move it to [RFC] and stall it if no activity by few days.

**Pull request closed without merging**